code/datums/components/supermatter_crystal.dm

```diff
@@ -152,9 +160,7 @@
 	var/atom/atom_source = source
 	if(!istype(item) || (item.item_flags & ABSTRACT) || !istype(user))
 		return
-	if(istype(item, /obj/item/melee/roastingstick))
-		return FALSE
-	if(istype(item, /obj/item/toy/crayon/spraycan))
+	if(is_type_in_typecache(item, attackby_hit_typecache))
```

Comment:
This would skip the custom cigarette code that happens later in the proc.  So cigarettes shouldn't be in the typecache list.

Reply:
smart idea lol

Reply:
also why did you close this?

Reply:
I pulled the 30-something commits and then that whole commit bloat happened and so I panicked and closed it. I just remade it